controllers/vaultdynamicsecret_controller.go

```diff
@@ -288,3 +304,55 @@ func isLeaseNotfoundError(err error) bool {
 	}
 	return false
 }
+
+// handleDeletion will handle the deletion path of the VDS secret:
+// * revoking any associated outstanding leases
+// * removing our finalizer
+func (r *VaultDynamicSecretReconciler) handleDeletion(ctx context.Context, o *secretsv1alpha1.VaultDynamicSecret) error {
+	logger := log.FromContext(ctx)
+	if err := r.revokeLease(ctx, o, ""); err != nil {
+		// Do not return error, otherwise we may fail to remove the finalizer.
+		// Worst case at this point we will leave a dangling lease instead of a secret which
+		// cannot be deleted.
+	}
+	if controllerutil.ContainsFinalizer(o, vaultDynamicSecretFinalizer) {
+		logger.Info("Removing finalizer")
+		if controllerutil.RemoveFinalizer(o, vaultDynamicSecretFinalizer) {
+			if err := r.Update(ctx, o); err != nil {
+				logger.Error(err, "Failed to remove the finalizer")
+				return err
+			}
+			logger.Info("Successfully removed the finalizer")
+		}
+	}
+	return nil
+}
+
+// revokeLease revokes the VDS secret's lease.
+// NOTE: Enabling revocation requires the VaultAuthMethod referenced by `o.Spec.VaultAuthRef` to have a policy
+// that includes `path "sys/leases/revoke" { capabilities = ["update"] }`, otherwise this will fail with permission
+// errors.
+func (r *VaultDynamicSecretReconciler) revokeLease(ctx context.Context, o *secretsv1alpha1.VaultDynamicSecret, id string) error {
+	logger := log.FromContext(ctx)
+	logger.Info("Revoking lease for credential ", "id", o.Status.SecretLease.ID)
```

Comment:
When `id != ""`, this ends up logging that the new credential is going to be revoked, instead of the old one, so we should probably define `leaseID` before and use it in this log call.